src/lb_data.c

```diff
@@ -1192,6 +1405,44 @@ int lb_halo_create(const lb_t * lb, lb_halo_t * h, lb_halo_enum_t scheme) {
     h->request[ireq] = MPI_REQUEST_NULL;
   }
 
+
+  /* Device */
+
+  int ndevice;
+  tdpGetDeviceCount(&ndevice);
+  tdpStreamCreate(&h->stream);
+
+  if (ndevice == 0) {
+    h->target = h;
+  }
+  else {
+    tdpAssert( tdpMalloc((void **) &h->target, sizeof(lb_halo_t)) );
+    tdpAssert( tdpMemcpy(h->target, h, sizeof(lb_halo_t),
+			 tdpMemcpyHostToDevice) );
```

Comment:
It might be a good idea to ensure the target structure is initialised to zero, e.g., with `cudaMemset()`.